### PR TITLE
Move HostConcatResultUtil out of unshimmed classes

### LINKED ISSUE #5513: [BUG] NoClassDefFoundError with caller classloader off in GpuShuffleCoalesceIterator in local-cluster
**Describe the bug**
running `test_cartesian_join_special_case_count` fails with 
`Caused by: java.lang.NoClassDefFoundError: com/nvidia/spark/rapids/Arm`
<details><pre>
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.$anonfun$next$3(GpuShuffleCoalesceExec.scala:214)
E                       at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
E                       at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.withResource(GpuShuffleCoalesceExec.scala:191)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.$anonfun$next$2(GpuShuffleCoalesceExec.scala:213)
E                       at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
E                       at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.withResource(GpuShuffleCoalesceExec.scala:191)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.$anonfun$next$1(GpuShuffleCoalesceExec.scala:207)
E                       at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
E                       at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.withResource(GpuShuffleCoalesceExec.scala:191)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.next(GpuShuffleCoalesceExec.scala:206)
E                       at com.nvidia.spark.rapids.GpuShuffleCoalesceIterator.next(GpuShuffleCoalesceExec.scala:191)
E                       at com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:283)
E                       at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:238)
E                       at scala.Option.getOrElse(Option.scala:189)
E                       at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:235)
E                       at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:181)
E                       at com.nvidia.spark.rapids.ColumnarToRowIterator.$anonfun$fetchNextBatch$2(GpuColumnarToRowExec.scala:241)
E                       at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
E                       at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
E                       at com.nvidia.spark.rapids.ColumnarToRowIterator.withResource(GpuColumnarToRowExec.scala:187)
E                       at com.nvidia.spark.rapids.ColumnarToRowIterator.fetchNextBatch(GpuColumnarToRowExec.scala:238)
E                       at com.nvidia.spark.rapids.ColumnarToRowIterator.loadNextBatch(GpuColumnarToRowExec.scala:215)
E                       at com.nvidia.spark.rapids.ColumnarToRowIterator.hasNext(GpuColumnarToRowExec.scala:255)
E                       at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
E                       at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:349)
E                       at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
E                       at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
E                       at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
E                       at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
E                       at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
E                       at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
E                       at org.apache.spark.scheduler.Task.run(Task.scala:131)
E                       at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
E                       at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1462)
E                       at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
E                       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
E                       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
E                       ... 1 more
</pre></details>
 

**Steps/Code to reproduce bug**
invoke:
```bash
TEST_PARALLEL=0 \
  PYSP_TEST_spark_rapids_force_caller_classloader=false \
  NUM_LOCAL_EXECS=1 \
  ./integration_tests/run_pyspark_from_build.sh -k test_cartesian_join_special_case_count
```

**Expected behavior**
Should pass

**Environment details (please complete the following information)**
 - Environment location: Standalone, 
 
 - **Additional context**

originally reported by h/t @pxLi

### PR DESCRIPTION
**Pull request by gerashegalov**
Don't use ai.rapidsai.cudf package for spark-rapids Scala classes. Otherwise it is going to be loaded by conventional classloader and fail to load referenced classes out of the shimmed areas.
- Move the class 
- add a smoking test to prevent this sort of regressions in premerge

Closes #5513. 

Depends on rapidsai/cudf#10949
    
Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala [hunk 1]
```diff
@@ -14,9 +14,9 @@
  * limitations under the License.
  */
 
-package ai.rapids.cudf
+package com.nvidia.spark.rapids.cudf_utils
 
-import ai.rapids.cudf.JCudfSerialization.HostConcatResult
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:19]: Nit: Not sure why this change was made, as it made this PR significantly larger than it needed to be.

**Review by abellina — DISMISSED** [review]
Thanks for fixing this @gerashegalov

**Review by jlowe — APPROVED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala (hunk 1 above)
Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:19]: mostly for consistency with how JCudfSerialization.SerializedTableHeader is referenced

Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:19]: mostly for consistency with how JCudfSerialization.SerializedTableHeader is referenced

**Review by jlowe — APPROVED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala [hunk 2]
```diff
@@ -28,8 +29,7 @@ object HostConcatResultUtil extends Arm {
    */
   def rowsOnlyHostConcatResult(numRows: Int): HostConcatResult = {
     new HostConcatResult(
-      new JCudfSerialization.SerializedTableHeader(
-        Array.empty, numRows, 0L),
+      new JCudfSerialization.SerializedTableHeader(numRows),
```
Comment (razajafri) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:32]: This constructor is not accessible now that this file is in a different package

Comment (razajafri) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:32]: This constructor is not accessible now that this file is in a different package

Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/cudf_utils/HostConcatResultUtil.scala:32]: Did you look at the dependency PR https://github.com/rapidsai/cudf/pull/10949

**Comment by gerashegalov** [comment]
build

**Pull request merged**